components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -291,22 +294,65 @@ public void updateAPI(API.APIBuilder apiBuilder) throws APIManagementException {
     @Override
     public void updateAPIStatus(String apiId, String status, Map<String, Boolean> checkListItemMap) throws
             APIManagementException {
+        boolean requireReSubscriptions = false;
+        boolean deprecateOlderVersion = false;
+        List<Subscription> skippedSubscriptionList = new ArrayList<>();
         try {
             API api = getApiDAO().getAPI(apiId);
             if (api != null) {
                 API.APIBuilder apiBuilder = new API.APIBuilder(api);
                 apiBuilder.lifecycleState(getApiLifecycleManager().getCurrentLifecycleState(apiBuilder
                         .getLifecycleInstanceId()));
                 for (Map.Entry<String, Boolean> checkListItem : checkListItemMap.entrySet()) {
-                    apiBuilder.lifecycleState(getApiLifecycleManager().checkListItemEvent(api.getLifecycleInstanceId
-                            (), status, checkListItem.getKey(), checkListItem.getValue()));
+                    if (APIMgtConstants.DEPRECATE_PREVIOUS_VERSIONS.equals(checkListItem.getKey())) {
+                        deprecateOlderVersion = checkListItem.getValue();
+                    } else if (APIMgtConstants.REQUIRE_RE_SUBSCRIPTIONS.equals(checkListItem.getKey())) {
+                        requireReSubscriptions = checkListItem.getValue();
+                    } else {
+                        apiBuilder.lifecycleState(getApiLifecycleManager().checkListItemEvent(api.getLifecycleInstanceId
+                                (), status, checkListItem.getKey(), checkListItem.getValue()));
+                    }
                 }
                 API originalAPI = apiBuilder.build();
                 getApiLifecycleManager().executeLifecycleEvent(status, apiBuilder
                         .getLifecycleInstanceId(), getUsername(), originalAPI);
+                if (deprecateOlderVersion) {
+                    if (StringUtils.isNotEmpty(api.getPreviousId())) {
+                        API oldAPI = getApiDAO().getAPI(api.getPreviousId());
+                        if (oldAPI != null) {
+                            API.APIBuilder previousAPI = new API.APIBuilder(oldAPI);
+                            previousAPI.setLifecycleStateInfo(getApiLifecycleManager().getCurrentLifecycleState
```

Comment:
What's the reason for setting the current life cycle state again just before deprecating it?